docs/integrations/development_guide.md

```diff
@@ -0,0 +1,576 @@
+
+Let's work on getting our DigitalOcean Account ingested.
+
+In `src/types.ts` we'll setup the basic type for our `DigitalOceanAccount`.
+
+```diff
+- export interface AcmeAccount {
+-  id: string;
+-  name: string;
++ export interface DigitalOceanAccount {
++  uuid: string;
++  email: string;
++  email_verified: boolean;
++  status: string;
++  status_message: string;
+}
+```
+
+Let's also make a change in `src/constants.ts` so our step knows information
+about the entity we are ingesting.
+
+```diff
+
+
+
+export const Entities: Record<'ACCOUNT', StepEntityMetadata> = {
+  ACCOUNT: {
+    resourceName: 'Account',
+-   _type: 'acme_account',
++   _type: 'digital_ocean_account',
+    _class: ['Account'],
+    schema: {
+      properties: {},
+      required: [],
+    },
+  },
+};
+```
+
+We'll also need to use a http client to make requests to our provider. Here
+we'll use the `integration-sdk-http-client`.
+
+```sh
+yarn add @jupiterone/integration-sdk-http-client
+```
+
+Edits to the `src/client.ts`
+
+# TODO: This is more edits than I really want :/
+
+```diff
+ import { IntegrationConfig } from './config';
+-import { AcmeAccount } from './types';
++import { DigitalOceanAccount } from './types';
++
++import { APIClient as sdkApiClient } from '@jupiterone/integration-sdk-http-client';
++import {
++  APIRequest,
++  APIResponse,
++} from '@jupiterone/integration-sdk-http-client/dist/src/types';
+
+ export type ResourceIteratee<T> = (each: T) => Promise<void> | void;
+
+@@ -12,22 +18,33 @@ export type ResourceIteratee<T> = (each: T) => Promise<void> | void;
+  * resources.
+  */
+ export class APIClient {
+-  constructor(readonly config: IntegrationConfig) {}
++  private client: sdkApiClient;
++
++  private BASE_URL = 'https://api.digitalocean.com/v2';
++
++  constructor(readonly config: IntegrationConfig) {
++    this.client = new sdkApiClient();
++  }
+
+   /**
+    * getAccount fetches the account details from the provider API.
+    * @returns {Promise<AcmeAccount>} A promise for the AcmeAccount
+    */
+-  public async getAccount(): Promise<AcmeAccount> {
+-    // This is where an authenticated request to the provider API would be made
+-    // We return a promise that resolves to mock data
+-    // to simulate an API response
+-    return new Promise<AcmeAccount>((resolve) => {
+-      resolve({
+-        id: 'account-id',
+-        name: 'Account Name',
+-      });
+-    });
++  public async getAccount(): Promise<DigitalOceanAccount> {
++    const request: APIRequest = {
++      url: this.BASE_URL + '/account',
++      headers: {
++        Authorization: `Bearer ${this.config.accessToken}`,
++      },
++      method: 'GET',
++    };
++    const response = await this.client.executeAPIRequest(request);
++    return response.data as DigitalOceanAccount;
++
++    // const response = await this.client.executeAPIRequest<DigitalOceanAccount>(
```

Comment:
Is this section commented out because our HTTP client does support response typing?

Reply:
I'm honestly not sure why that's there. I'm going to delete it :+1: